diff --git a/src/luaconf.h b/src/luaconf.h
index c2d29d94..909f0d80 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -20,10 +20,23 @@
 */
 #define LUA_LDIR	"!\\lua\\"
 #define LUA_CDIR	"!\\"
+#define LUA_SHRDIR	"!\\..\\share\\lua\\5.1\\"
+
 #define LUA_PATH_DEFAULT \
-  ".\\?.lua;" LUA_LDIR"?.lua;" LUA_LDIR"?\\init.lua;"
+  "!\\..\\share\\luajit-2.1.0-beta3\\?.lua;" \
+  LUA_LDIR"?.lua;" \
+  LUA_LDIR"?\\init.lua;" \
+  LUA_CDIR"?.lua;" \
+  LUA_CDIR"?\\init.lua;" \
+  LUA_SHRDIR"?.lua;" \
+  LUA_SHRDIR"?\\init.lua;" \
+  ".\\?.lua;" \
+  ".\\?\\init.lua"
 #define LUA_CPATH_DEFAULT \
-  ".\\?.dll;" LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
+  LUA_CDIR"?.dll;" \
+  LUA_CDIR"..\\lib\\lua\\5.1\\?.dll;" \
+  LUA_CDIR"loadall.dll;" \
+  ".\\?.dll"
 #else
 /*
 ** Note to distribution maintainers: do NOT patch the following lines!
